### PR TITLE
Add ignore.txt for 2.12

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY

https://github.com/ansible-collections/overview/issues/45#issuecomment-813821100

Depends-On: https://github.com/ansible-collections/junipernetworks.junos/issues/165

### CODE REVIEW

**Comment by NilashishC** [comment]
@Andersson007 Thank you for this PR! A trivial changelog fragment is needed for CI to pass. Could you please add that too?

**Comment by Andersson007** [comment]
@NilashishC done, thanks

File: changelogs/fragments/170-add_ignore_txt.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+trivial:
+- Add ingore-2.12.txt.
```
Comment (Andersson007) [inline changelogs/fragments/170-add_ignore_txt.yml:3]:
```suggestion
  - Add ingore-2.12.txt.
```

**Comment by Andersson007** [comment]
@NilashishC i fixed linter issues also. So it's ready for review. I don't think the other errors are related to the PR directly

File: changelogs/fragments/170-add_ignore_txt.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+trivial:
+  - Add ingore-2.12.txt.
```
Comment (rohitthakur2590) [inline changelogs/fragments/170-add_ignore_txt.yml:3]: This Looks to be a typo.

Comment (Andersson007) [inline changelogs/fragments/170-add_ignore_txt.yml:3]:
```suggestion
  - Add ignore-2.12.txt.
```

Comment (Andersson007) [reply changelogs/fragments/170-add_ignore_txt.yml]: @rohitthakur2590 fixed thanks. Aren't trivial fragments ignored anyway?

Comment (Andersson007) [reply changelogs/fragments/170-add_ignore_txt.yml]: I'm not sure about this collection, so interesting.

**Review by rohitthakur2590 — APPROVED** [review]

**Comment by Andersson007** [comment]
```
ERROR: Found 16 validate-modules issue(s) which need to be resolved:
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key' in argument_spec could be a secret, though doesn't have `no_log` set found in config
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups -> neighbors
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups -> neighbors
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'expedited_update_tokens' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> output_queue_priority
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> bfd_liveness_detection -> authentication
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups -> bfd_liveness_detection -> authentication
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups -> neighbors -> bfd_liveness_detection -> authentication
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'priority_update_tokens' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> output_queue_priority
ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'update_tokens' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> output_queue_priority -> priority_update_tokens
ERROR: plugins/modules/junos_ospf_interfaces.py:0:0: no-log-needed: Argument 'key_value' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> address_family -> processes -> authentication -> md5
ERROR: plugins/modules/junos_ospf_interfaces.py:0:0: no-log-needed: Argument 'simple_password' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> address_family -> processes -> authentication
ERROR: plugins/modules/junos_user.py:0:0: no-log-needed: Argument 'sshkey' in argument_spec could be a secret, though doesn't have `no_log` set
ERROR: plugins/modules/junos_user.py:0:0: no-log-needed: Argument 'sshkey' in argument_spec could be a secret, though doesn't have `no_log` set found in aggregate
```
I think it shouldn't be fixed in scope of this PR

**Comment by rohitthakur2590** [comment]
> ```
> ERROR: Found 16 validate-modules issue(s) which need to be resolved:
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key' in argument_spec could be a secret, though doesn't have `no_log` set found in config
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups -> neighbors
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'authentication_key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups -> neighbors
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'expedited_update_tokens' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> output_queue_priority
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> bfd_liveness_detection -> authentication
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups -> bfd_liveness_detection -> authentication
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'key_chain' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> groups -> neighbors -> bfd_liveness_detection -> authentication
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'priority_update_tokens' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> output_queue_priority
> ERROR: plugins/modules/junos_bgp_global.py:0:0: no-log-needed: Argument 'update_tokens' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> output_queue_priority -> priority_update_tokens
> ERROR: plugins/modules/junos_ospf_interfaces.py:0:0: no-log-needed: Argument 'key_value' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> address_family -> processes -> authentication -> md5
> ERROR: plugins/modules/junos_ospf_interfaces.py:0:0: no-log-needed: Argument 'simple_password' in argument_spec could be a secret, though doesn't have `no_log` set found in config -> address_family -> processes -> authentication
> ERROR: plugins/modules/junos_user.py:0:0: no-log-needed: Argument 'sshkey' in argument_spec could be a secret, though doesn't have `no_log` set
> ERROR: plugins/modules/junos_user.py:0:0: no-log-needed: Argument 'sshkey' in argument_spec could be a secret, though doesn't have `no_log` set found in aggregate
> ```
> 
> I think it shouldn't be fixed in scope of this PR

Thank you for the required changes, yes there is WIP for no_log failures, after that we are good to merge this PR.

**Comment by pabelanger** [comment]
Adding depends-on to the no_log fix should allow this to pass.

**Comment by Andersson007** [comment]
Depends-On: https://github.com/ansible-collections/junipernetworks.junos/issues/173

**Comment by Andersson007** [comment]
recheck

**Comment by Andersson007** [comment]
@pabelanger doesn't it work with issues?
I put Depends-On: https://github.com/ansible-collections/junipernetworks.junos/issues/165 and `recheck` and the same result

**Comment by rohitthakur2590** [comment]
closing as this is resolved by https://github.com/ansible-collections/junipernetworks.junos/pull/164

**Comment by Andersson007** [comment]
great, thanks everyone!

**Pull request closed without merging**